### PR TITLE
[core] fix installing expo hostobject thread issue on ios hermes

### PR DESCRIPTION
**Pull request by Kudo**
# Why

Hermes is much thread sensitive, there are some crashes if we don't setup js stuffs on js thread.
possible fix for https://github.com/realm/realm-js/issues/4735

# How

install expo host object in js thread (originally it' in another worker thread)

# Test Plan

- ci passed
- try realm repro in https://github.com/realm/realm-js/issues/4735

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by carbopilot** [comment]
Brilliant work Kudo! Thanks!

**Comment by DaBs** [comment]
Great fix - I'd consider changing the changelog entry to mention it being other JSI modules instead of only Realm, so other React Native module authors can have a good idea that this change may also be the fix to any users reporting issues to them in user-land.

Again, thanks for picking this up so quickly! 🚀

File: packages/expo-modules-core/CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
 - Fixed libraries using the `ViewDefinitionBuilder` crashes when ProGuard or R8 is enabled on Android. ([#20197](https://github.com/expo/expo/pull/20197) by [@lukmccall](https://github.com/lukmccall))
 - Fixed Either types not supporting non-primitive types on iOS. ([#20247](https://github.com/expo/expo/pull/20247) by [@tsapeta](https://github.com/tsapeta))
 - Fixed Function not supporting certain arities on Android. ([#20419](https://github.com/expo/expo/pull/20419) by [@motiz88](https://github.com/motiz88))
+- Fixed threading crash issue when running with Realm and Hermes on iOS. ([#20506](https://github.com/expo/expo/pull/20506) by [@kudo](https://github.com/kudo))
```
Comment (Kudo) [inline packages/expo-modules-core/CHANGELOG.md:22]:
```suggestion
- Fixed threading crash issue when running with Hermes on iOS. ([#20506](https://github.com/expo/expo/pull/20506) by [@kudo](https://github.com/kudo))
```

**Review by lukmccall — APPROVED** [review]
Looks good to me 🔥

**Review by tsapeta — APPROVED** [review]

**Pull request merged**